bootstrap/v2/pkg/apis/apps/kfdef/v1alpha1/application_types.go

```diff
@@ -15,30 +15,49 @@
 package v1alpha1
 
 import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	gogetter "github.com/hashicorp/go-getter"
 	"github.com/kubeflow/kubeflow/bootstrap/config"
+	kfapis "github.com/kubeflow/kubeflow/bootstrap/pkg/apis"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
 	"k8s.io/api/v2/core/v1"
 	metav1 "k8s.io/apimachinery/v2/pkg/apis/meta/v1"
+	"os"
+	"path"
 )
 
+const KfConfigFile = "app.yaml"
+
 // KfDefSpec holds common attributes used by each platform
 type KfDefSpec struct {
 	config.ComponentConfig `json:",inline"`
-	AppDir                 string `json:"appdir,omitempty"`
-	Version                string `json:"version,omitempty"`
-	MountLocal             bool   `json:"mountLocal,omitempty"`
-	Project                string `json:"project,omitempty"`
-	Email                  string `json:"email,omitempty"`
-	IpName                 string `json:"ipName,omitempty"`
-	Hostname               string `json:"hostname,omitempty"`
-	Zone                   string `json:"zone,omitempty"`
-	UseBasicAuth           bool   `json:"useBasicAuth"`
-	SkipInitProject        bool   `json:"skipInitProject,omitempty"`
-	UseIstio               bool   `json:"useIstio"`
-	EnableApplications     bool   `json:"enableApplications"`
-	ServerVersion          string `json:"serverVersion,omitempty"`
-	DeleteStorage          bool   `json:"deleteStorage,omitempty"`
-	PackageManager         string `json:"packageManager,omitempty"`
-	ManifestsRepo          string `json:"manifestsRepo,omitempty"`
+	// TODO(jlewi): Why is AppDir a part of the spec? AppDir is currently used
+	// to refer to the location on disk where all the manifests are stored.
+	// But that should be treated as a local cache and not part of the spec.
+	// For example, if the app is checked out on a different machine the AppDir will change.
+	// AppDir. AppDir is stored in KfDefSpec because we pass a KfDef around to
+	// as a way to pass the information to all the KfApps that need to know the local AppDir.
+	// A better solution might be to store AppDir in KfDef.Status to better reflect its
+	// ephemeral nature and match K8s semantics.
```

Comment:
if we move KfDef to Application then these fields can be in ApplicationSpec.Info which is a [[]InfoItem](https://github.com/kubernetes-sigs/application/blob/master/pkg/apis/app/v1beta1/application_types.go#L92).
Each InfoItem's ValueFrom could point to a ConfigMap entry.

Reply:
Thanks. Maybe open an issue for that? I explicitly decided not to touch AppDir in this PR (or probably 0.6) because I was trying to minimize churn.